client/relay-chain-rpc-interface/src/reconnecting_ws_client.rs

```diff
@@ -0,0 +1,433 @@
+use cumulus_primitives_core::relay_chain::Header as PHeader;
+use cumulus_relay_chain_interface::{RelayChainError, RelayChainResult};
+use futures::{
+	channel::{
+		mpsc::{Receiver, Sender},
+		oneshot::Sender as OneshotSender,
+	},
+	stream::FuturesUnordered,
+	StreamExt,
+};
+use jsonrpsee::{
+	core::{
+		client::{Client as JsonRpcClient, ClientT, Subscription, SubscriptionClientT},
+		Error as JsonRpseeError, JsonValue,
+	},
+	ws_client::WsClientBuilder,
+};
+use polkadot_service::TaskManager;
+use std::{future::Future, sync::Arc};
+use tokio::sync::mpsc::{
+	channel as tokio_channel, error::TryRecvError, Receiver as TokioReceiver, Sender as TokioSender,
```

Comment:
I use it here because tokio channels require only [`&self`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#method.send) on send compared to [`&mut self` ](https://docs.rs/futures/latest/futures/channel/mpsc/struct.Sender.html#method.try_send)for the `futures` channels. This is much more convenient here.